agent/agent.go

```diff
@@ -313,22 +317,103 @@ type Agent struct {
 	// IP.
 	httpConnLimiter connlimit.Limiter
 
+	// Connection Pool
+	connPool *pool.ConnPool
```

Comment:
The Agent code now needs to make insecure RPCs to servers for autoconfig so the shared connection pool has been moved up a level into the Agent and we pass it down into the Server/Client